omod/src/main/java/org/openmrs/module/dhisreport/web/controller/ReportDefinitionController.java

```diff
@@ -143,4 +150,62 @@ public String editDataValueTemplate( ModelMap model,
             + dvt.getReportDefinition().getId();
     }
 
+    @RequestMapping( value = "/module/dhisreport/getReportDefinitions", method = RequestMethod.POST )
+    public String getReportDefinitions( WebRequest webRequest, HttpServletRequest request )
+    {
+        String username = Context.getAdministrationService().getGlobalProperty( "dhisreport.dhis2UserName" );
+        String password = Context.getAdministrationService().getGlobalProperty( "dhisreport.dhis2Password" );
+        String dhisurl = Context.getAdministrationService().getGlobalProperty( "dhisreport.dhis2URL" );
+        String url = dhisurl + "/api/dataSets";
+        //String url = "https://play.dhis2.org/demo/api/dataSets";
+        String referer = webRequest.getHeader( "Referer" );
+        HttpSession session = request.getSession();
+        
+        try
+        {
+            DefaultHttpClient httpClient = new DefaultHttpClient();
+            HttpGet getRequest = new HttpGet( url );
+            getRequest.addHeader( "accept", "application/dsd+xml" );
+            getRequest.addHeader( BasicScheme.authenticate( new UsernamePasswordCredentials( username, password ),
+                "UTF-8", false ) );
+            HttpResponse response = httpClient.execute( getRequest );
+
+            if ( response.getStatusLine().getStatusCode() != 200 )
```

Comment:
i am keeping this code for debugging purposes only. Please let me know if we don't need it..